service/scaladsl/client/src/main/scala/com/lightbend/lagom/scaladsl/client/ServiceClient.scala

```diff
@@ -141,31 +148,162 @@ trait LagomServiceClientComponents extends TopicFactoryProvider { self: LagomCon
+ *   extends LagomClientFactory("my-client")
+ *   with StaticServiceLocatorComponents
+ *   with AhcWSComponents {
+ *
+ *   override def staticServiceUri = URI.create("http://localhost:8080")
+ * }
+ *
+ *
+ * val actorSystem = ActorSystem("my-app")
+ * val materializer = ActorMaterializer()(actorSystem)
+ * val clientFactory = new MyLagomClientFactory(actorSystem, materializer)
+ * }}}
+ *
+ * @param clientName The name of the service that is consuming the Lagom service. This will impact how calls made through clients
+ *                   generated by this factory will identify themselves.
+ * @param classLoader A classloader, it will be used to create the service proxy and needs to have the API for the client in it.
+ */
+abstract class LagomClientFactory(
+  clientName:  String,
+  classLoader: ClassLoader = classOf[LagomClientFactory].getClassLoader
 ) extends LagomServiceClientComponents with LagomConfigComponent {
+
   private val defaultApplicationLifecycle = new DefaultApplicationLifecycle
 
   override lazy val serviceInfo: ServiceInfo = ServiceInfo(clientName, Map.empty)
   override lazy val environment: Environment = Environment(new File("."), classLoader, Mode.Prod)
+
   lazy val configuration: Configuration = Configuration.load(
     environment.classLoader,
     System.getProperties,
     Map.empty,
     allowMissingApplicationConf = true
   )
-  override lazy val applicationLifecycle: ApplicationLifecycle = defaultApplicationLifecycle
-  lazy val actorSystem: ActorSystem = new ActorSystemProvider(environment, configuration).get
-  lazy val coordinatedShutdown: CoordinatedShutdown = new CoordinatedShutdownProvider(actorSystem, applicationLifecycle).get
 
-  override lazy val materializer: Materializer = ActorMaterializer.create(actorSystem)
+  override lazy val applicationLifecycle: ApplicationLifecycle = defaultApplicationLifecycle
   override lazy val executionContext: ExecutionContext = actorSystem.dispatcher
 
   /**
-   * Stop the application.
+   * Override this method if your [[LagomClientFactory]] implementation needs to free any resource.
+   *
+   * For example, when implementing your own [[LagomClientFactory]], you may choose to reuse an existing [[akka.actor.ActorSystem]],
+   * but use a internal [[akka.stream.Materializer]]. In which case, you can use this method to only shutdown the [[akka.stream.Materializer]].
+   *
+   * If you override this method, make sure you also release the internally managed resources
+   * by calling [[#releaseInternalResources()]] method.
+   *
    */
-  def stop(): Unit = CoordinatedShutdownSupport.syncShutdown(actorSystem, ClientStoppedReason)
+  def stop(): Unit = {
+    Await.result(
+      releaseInternalResources(),
+      Duration(5, TimeUnit.SECONDS)
+    )
+  }
+
+  protected def releaseInternalResources(): Future[Done] =
+    defaultApplicationLifecycle.stop().map(_ => Done)(executionContext)
```

Comment:
I'm explicitly choosing now to keep `defaultApplicationLifecycle` private and stop it through this new protected method. If we find a better solution later, we can redefine its impl or make it a noops.